### PR TITLE
CLN: Old string formatting: .format() -> f""

### PR DESCRIPTION
**Pull request by baevpetr**
- [x] xref #29547
- [x] passes `black pandas`
Old style % formatting to f-strings

### CODE REVIEW

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -1056,8 +1046,7 @@ def coerce_to_target_dtype(self, other):
                 return self.astype(object)
 
             raise AssertionError(
-                "possible recursion in "
-                "coerce_to_target_dtype: {} {}".format(self, other)
+                "possible recursion in " f"coerce_to_target_dtype: {self} {other}"
```
Comment (jbrockmendel) [inline pandas/core/internals/blocks.py]: an extra space snuck in the middle here (this is a common occurrence with black)

File: pandas/core/internals/blocks.py [hunk 2]
```diff
@@ -1202,8 +1191,7 @@ def _interpolate(
         if method in ("krogh", "piecewise_polynomial", "pchip"):
             if not index.is_monotonic:
                 raise ValueError(
-                    "{0} interpolation requires that the "
-                    "index be monotonic.".format(method)
+                    f"{method} interpolation requires that the " "index be monotonic."
```
Comment (jbrockmendel) [inline pandas/core/internals/blocks.py]: extra space

File: pandas/core/internals/blocks.py [hunk 3]
```diff
@@ -544,15 +541,15 @@ def astype(self, dtype, copy: bool = False, errors: str = "raise"):
...
+                f"Expected value of kwarg 'errors' to be one of {list(errors_legal_values)}. "
+                f"Supplied value is '{errors}'"
             )
             raise ValueError(invalid_arg)
 
         if inspect.isclass(dtype) and issubclass(dtype, ExtensionDtype):
             msg = (
-                "Expected an instance of {}, but got the class instead. "
-                "Try instantiating 'dtype'.".format(dtype.__name__)
+                f"Expected an instance of {dtype.__name__}, but got the class instead. "
```
Comment (jbrockmendel) [inline pandas/core/internals/blocks.py]: looks like the linter thinks this line is too long

**Comment by jbrockmendel** [comment]
Couple of small issues, otherwise this is a nice cleanup, thanks

File: pandas/core/indexing.py [hunk 4]
```diff
@@ -1905,17 +1900,14 @@ def _validate_key(self, key, axis: int):
...
+                raise IndexError(f".iloc requires numeric indexers, got {arr}")
 
             # check that the key does not exceed the maximum size of the index
             if len(arr) and (arr.max() >= len_axis or arr.min() < -len_axis):
                 raise IndexError("positional indexers are out-of-bounds")
         else:
             raise ValueError(
-                "Can only index by location with "
-                "a [{types}]".format(types=self._valid_types)
+                "Can only index by location with " f"a [{self._valid_types}]"
```
Comment (jbrockmendel) [inline pandas/core/indexing.py]: extra space crept in, also down on 2059 (im going to stop pointing those out now)

File: pandas/core/indexing.py [hunk 5]
```diff
@@ -1167,16 +1164,14 @@ def _validate_read_indexer(
         if missing:
             if missing == len(indexer):
                 raise KeyError(
-                    "None of [{key}] are in the [{axis}]".format(
-                        key=key, axis=self.obj._get_axis_name(axis)
-                    )
+                    f"None of [{key}] are in the [{self.obj._get_axis_name(axis)}]"
```
Comment (jbrockmendel) [inline pandas/core/indexing.py]: nitpick: can you define `axis_name = self.obj._get_axis_name(axis)` on the previous line and use `axis_name` inside the fstring?

File: pandas/core/indexing.py (hunk 4 above)
Comment (baevpetr) [reply pandas/core/indexing.py]: My fault. Missed them.

File: pandas/core/indexing.py (hunk 5 above)
Comment (baevpetr) [reply pandas/core/indexing.py]: Done.

**Comment by baevpetr** [comment]
So there're some more `" f" ` typos.

**Review by simonjayhawkins — APPROVED** [review]
@baevpetr lgtm.

**Comment by jreback** [comment]
thanks @baevpetr

**Pull request merged**